### DISCUSSION TITLE
Remove or Remap Existing Routes Dynamically

### DISCUSSION THREAD

**Discussion by sirinath** [activej/activej#189] [Q&A]
How can the routes be removed or remapped dynamically? If this not possible can this be added.

Especially when created with a standard launcher.

**Comment by eduard-vasinskyi** [comment]
Hello, @sirinath
You can dynamically change servlet that handles a request by choosing a required servlet in your code, based on your application logic. This approach is more explicit and flexible than configuring routes via XML or any other external configuration.

All you need to do is to inspect parameters of incoming HTTP request and choose a desired servlet for further request processing.
Here are some snippets that may guide you:
- [Routing based on host](https://gist.github.com/eduard-vasinskyi/608db17cc37b01888bdac0ad4d6b8dd4)
- [Routing based on mime type](https://gist.github.com/eduard-vasinskyi/11b0774b317ff26483d47be1ecbd0aac)
- [Routing based on cookie](https://gist.github.com/eduard-vasinskyi/d0073ccca5ce5275ec11eb34605ac788)
- [Randomized routing](https://gist.github.com/eduard-vasinskyi/17ea7ddb4dbbb00a904ca40c9b27f25f)
